src/gwt/src/org/rstudio/studio/client/server/remote/ClientEventDispatcher.java

```diff
@@ -214,11 +214,13 @@ else if (type.equals(ClientEvent.ConsoleOutput))
          {
             ConsoleText output = event.getData();
             eventBus_.fireEvent(new ConsoleWriteOutputEvent(output));
+            eventBus_.fireEventToAllSatellites(new ConsoleWriteOutputEvent(output));
          }
          else if (type.equals(ClientEvent.ConsoleError))
          {
             ConsoleText error = event.getData();
             eventBus_.fireEvent(new ConsoleWriteErrorEvent(error));
+            eventBus_.fireEventToAllSatellites(new ConsoleWriteErrorEvent(error));
```

Comment:
The main window dispatches client events to satellites automatically:

https://github.com/rstudio/rstudio/blob/9c25b3adf4d58f314792bae255e606815bf9f1d6/src/gwt/src/org/rstudio/studio/client/common/satellite/SatelliteManager.java#L343

Is it possible that events are still being buffered for your particular satellite window (i.e. the manager isn't aware that it's done loading)?